### PR TITLE
Transform dialog

### PR DESCRIPTION
**Pull request by Lunalo**

### CODE REVIEW

**Comment by dannyparsons** [comment]
Great to see this dialog. Have the other team members tested it completely to find any bugs?

**Comment by Lunalo** [comment]
Hi Danny,

The dialog is still in progress

Regards,
John Ong'ala Lunalo|| Kenyatta University Alumnus ||
Volunteer, African Maths Initiative || Software Developer, African Data
Initiative ||
Email Address :johnlunalo95@gmail.com
Tel: +254701519572, +254786438572 || Skype contact: john.lunalo

**Comment by Lunalo** [comment]
Hi @dannyparsons  @bethanclarke  

I have tested this  dialog for  bugs.
You can review the dialog but am still looking for appropriate data to test for word function. 

Thanks.

**Comment by dannyparsons** [comment]
You can use any data that has character columns with some fairly long strings. Any data from a survey should do, just search online for csv data sets. Or membersurvey from surveydata for example.

**Comment by bethanclarke** [comment]
Hi John, good work so far. All the options are working well except word - more due to the command than anything, comments to come later for that.

"Convert Case" should have a default, so set the default to be "Lower".

For "Pad", remove the checkbox (we changed our mind on this) and add options for "Left", "Right" and "Both" - radio button for each. Then underneath this option put the "Pad" and then underneath that put the "Width". You should only be able to type in one character or number into the combo box for "Pad", otherwise you get an error (might need help from Danny to implement this for the Input control).

For "Trim", add options "Left", "Right" and "Both" - radio buttons for each for the "side" argument.

Underneath "Word", add an additional option, "Substring". This uses the command str_sub with: "To" and "From" numeric up downs for the start and end parameters. The default for "To" is 1 and for "End" is -1. You might want to look at the examples in the help for str_sub to see what the minus signs mean. So the numeric up downs can have negative numbers.
Finally, this dialogue doesn't write multiple columns, so the label shouldn't be Prefix. When you run the command again, the default column name should be the next available name e.g. New_Text1. The input control has an option to do this. 

Great work so far, this is a complicated dialog!

**Comment by bethanclarke** [comment]
Also, the dialogue doesn't open from form main

**Comment by Lunalo** [comment]
Hi Danny,

I didn't want to edit frmMain in this pull request.

That is why the dialog is not declared in the frmMain.

Thanks.

Regards,
John Ong'ala Lunalo|| Kenyatta University Alumnus ||
Volunteer, African Maths Initiative || Software Developer, African Data
Initiative ||
Email Address :johnlunalo95@gmail.com
Tel: +254701519572, +254786438572 || Skype contact: john.lunalo

**Comment by dannyparsons** [comment]
That's sensible. Do you want to do it in another pull request?

**Comment by Lunalo** [comment]
Hi @dannyparsons @bethanclarke 

This is now fixed

**Comment by bethanclarke** [comment]
I've had a quick look at the dialogue and have spotted a few things that are wrong with it. 

When I first run the dialogue, with the option "Convert Case" and "Lower" (default option), it doesn't work properly. 
There is a problem with the "Pad" radio buttons.
"Substring" isn't working properly - not all of the arguments are being passed into the command.

Test all of the options and look at the output R code to see if it matches up with options you've selected on the dialogue. Try switching between options before clicking okay.

File: instat/dlgTransformText.vb [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+            ucrBase.clsRsyntax.RemoveParameter("start")
+            ucrBase.clsRsyntax.RemoveParameter("end")
+            ucrBase.clsRsyntax.RemoveParameter("sep")
+
+        ElseIf rdoPad.Checked Then
+            ucrBase.clsRsyntax.SetFunction("stringr::str_pad")
+            PadSideParameter()
+            SeperatorParameter()
+            WidthParameter()
+            rdoLeft.Checked = True
```
Comment (dannyparsons) [inline instat/dlgTransformText.vb]: This should be in SetDefaults() only, not here as well.

File: instat/dlgTransformText.vb [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+            nudFirstWord.Visible = False
+            chkOrColumn.Visible = False
+            lblLastWord.Visible = False
+            nudLastWord.Visible = False
+            ucrInputSeparator.Visible = False
+            lblSeparator.Visible = False
+            rdoBoth.Visible = False
+            rdoRight.Visible = False
+            rdoLeft.Visible = False
+            rdoLeftTrim.Checked = True
```
Comment (dannyparsons) [inline instat/dlgTransformText.vb]: Should be in SetDefaults() only.

File: instat/dlgTransformText.vb [hunk 3]
```diff
@@ -0,0 +1,569 @@
...
+                ucrReceiverOrColumn.Visible = True
+                lblWidth.Visible = False
+                nudWidth.Visible = False
+                lblPad.Visible = False
+                ucrInputPad.Visible = False
+                ucrBase.clsRsyntax.RemoveParameter("pad")
+                ucrBase.clsRsyntax.RemoveParameter("width")
+                ucrBase.clsRsyntax.RemoveParameter("side")
+
+            ElseIf chkOrColumn.Checked = False
```
Comment (dannyparsons) [inline instat/dlgTransformText.vb]: You need "Then" at the end here.

**Pull request merged**